iceberg/src/main/scala/org/apache/spark/sql/delta/icebergShaded/HudiSchemaUtils.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.icebergShaded
+
+import org.apache.avro.{LogicalTypes, Schema}
+import org.apache.spark.sql.delta.metering.DeltaLogging
+import org.apache.spark.sql.types._
+
+import java.util
+import scala.collection.JavaConverters._
+
+object HudiSchemaUtils extends DeltaLogging {
+
+  /////////////////
+  // Public APIs //
+  /////////////////
+  def convertDeltaSchemaToHudiSchema(deltaSchema: StructType): Schema = {
+    /**
+     * Recursively (i.e. for all nested elements) transforms the delta DataType `elem` into its
+     * corresponding Avro type.
+     */
+    def transform[E <: DataType](elem: E, isNullable: Boolean): Schema = elem match {
+      case StructType(fields) =>
+
+        val avroFields: util.List[Schema.Field] = fields.map(f =>
+          new Schema.Field(
+            f.name,
+            transform(f.dataType, f.nullable),
+            f.getComment().orNull)).toList.asJava
+        finalizeSchema(
+          Schema.createRecord(elem.typeName, null, null, false, avroFields),
+          isNullable)
+
+      case ArrayType(elementType, containsNull) =>
```

Comment:
Uniform Iceberg actually supports List/Map now. I will leave it to u to add this support or not in this PR. if we decide not, I suggest add a check in OptimisticTransaction.prepareCommit to fail the txn if it enables Uniform Hudi on table with list/map in the schema. Otherwise this will throw during conversion and post commit which will be silent to the user.